source/lp_ticker.c

```diff
@@ -50,20 +53,33 @@ void lp_ticker_init(void) {
          * We want to use faster speeds, so we use 0.5ms tick for a max. of ~131MHz.
          * MAKE SURE `MINAR_PLATFORM_TIME_BASE` in the target configuration is set to `2000`!
          */
-        // Get clock configuration
-        RCC_ClkInitTypeDef RCC_ClkInitStruct;
-        uint32_t PclkFreq;
-        // Note: PclkFreq contains here the Latency (not used after)
-        HAL_RCC_GetClockConfig(&RCC_ClkInitStruct, &PclkFreq);
-        // Get TIM1 clock value
-        PclkFreq = HAL_RCC_GetPCLK1Freq();
-        // TIMxCLK = PCLKx when the APB prescaler = 1 else TIMxCLK = 2 * PCLKx
-        if (RCC_ClkInitStruct.APB1CLKDivider != RCC_HCLK_DIV1) {
-            PclkFreq *= 2;
-        }
-        TimMasterHandle.Instance               = TIM2;
+	// Get clock configuration
+	HAL_RCC_GetClockConfig(&RCC_ClkInitStruct, &TimClk);
```

Comment:
You can do it this way. However, I want to point out a couple of things:

1. You need to have different `MINAR_PLATFORM_TIME_BASE` values depending on the clock speed for the target
You can do it this way, but it is a maintenance overhead. As mentioned in the comment there are no STM32F* devices with a max clock speed of 131MHz (of clock prescaler frequency!!!). The F7 tops out at ~200MHz, which means ~100MHz for the prescaler clock.

2. `MINAR_PLATFORM_TIME_BASE` is a compile time constant. Lets assume it is fixed for the STM32F4xx to `2000` for a 0.5ms tick. Now you run the STM32F4 not at full clock speed, but at a lower speed, this algorithm would choose a 1ms tick, since it can. You can do it this way, but it is a maintenance overhead.

3. 1282Hz? See above.